src/app/pages/partners/analytics-events.js

```diff
@@ -1,52 +1,81 @@
+import React from 'react';
 import analytics from '~/helpers/analytics';
-import {books, types, advanced} from './store';
+import useSearchContext from './search-context';
 
-function toLabel(store) {
-    return store.value.join(',') || 'N/A';
+function toLabel(storeValue) {
+    return storeValue.join(',') || 'N/A';
 }
 
-function sendFilterEvent(name) {
-    const category = `Partner tool ${name}`;
+export function useValueChangeEvents() {
+    const {books, types, advanced} = useSearchContext();
+    const filterIsSelected = Boolean(types.size || advanced.size || books.size);
+    const sendFilterEvent = React.useCallback(
+        (name) => {
+            const category = `Partner tool ${name}`;
 
-    analytics.sendPageEvent(category, 'filter', toLabel(books));
-}
+            analytics.sendPageEvent(category, 'filter', toLabel(books.value));
+        },
+        [books]
+    );
+    const SAERef = React.useRef();
 
-function sendAddEvent(actionObj) {
-    if ('add' in actionObj) {
-        sendFilterEvent(actionObj.add);
-    }
-}
+    SAERef.current = React.useCallback(
```

Comment:
should it cause a re-render when `SAERef` is set?  if so you'll need to store it in state, ref assignment doesn't cause a re-render

Reply:
I'm using SAERef to avoid having `sendAddEvent` get re-called each time the `sendFilterEvent` callback updates.